src/main/java/org/folio/dataexp/service/export/ExportContext.java

```diff
@@ -0,0 +1,31 @@
+package org.folio.dataexp.service.export;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Component
+@RequiredArgsConstructor
+public class ExportContext {
+
+  private final AtomicBoolean lastSlice;
+  private final AtomicBoolean lastExport;
+
+  public void setLastSlice(boolean lastSlice) {
+    this.lastSlice.set(lastSlice);
```

Comment:
hi @obozhko-folio, as ExportContext is singleton here - setters could produce unexpected behavior for execution in several threads.